### PR TITLE
POC of randomized event generator for history events

### PR DESCRIPTION
**Pull request by yux0**
POC of randomized event generator for history events. This generator is to generate history events to test 3+DC logic.

### CODE REVIEW

File: host/xdc/event_generator.go [hunk 1]
```diff
@@ -0,0 +1,374 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
```
Comment (wxing1292) [inline host/xdc/event_generator.go]: i would put this in common/xdc

Comment (wxing1292) [reply host/xdc/event_generator.go]: and copyright 2019

File: host/xdc/event_generator.go [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+type (
+	// Model is a group of edges
+	Model interface {
+		AddEdge(...Edge)
+		ListEdges() []Edge
+	}
+
+	// Generator generates random events based on the defined models
+	Generator interface {
+		AddInitialEntryVertex(...Vertex)
```
Comment (wxing1292) [inline host/xdc/event_generator.go]: we might need more description about the terminology
let us sync on Monday

File: host/xdc/history_event_test.go [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+	ewfModel := NewHistoryEventModel()
+	ewfSignal := NewHistoryEvent(shared.EventTypeSignalExternalWorkflowExecutionInitiated.String())
+	ewfSignalFailed := NewHistoryEvent(shared.EventTypeSignalExternalWorkflowExecutionFailed.String())
+	ewfSignaled := NewHistoryEvent(shared.EventTypeExternalWorkflowExecutionSignaled.String())
+	ewfCancel := NewHistoryEvent(shared.EventTypeRequestCancelExternalWorkflowExecutionInitiated.String())
+	ewfCancelFail := NewHistoryEvent(shared.EventTypeRequestCancelExternalWorkflowExecutionFailed.String())
+	ewfCanceled := NewHistoryEvent(shared.EventTypeExternalWorkflowExecutionCancelRequested.String())
+
+	dtCompleteToEWFSignal := NewConnection(dtComplete, ewfSignal)
```
Comment (wxing1292) [inline host/xdc/history_event_test.go]: try use variable names that is more clear, like decision timeout? i guess for this case

File: host/xdc/event_generator.go (hunk 1 above)
Comment (yux0) [reply host/xdc/event_generator.go]: Updated the location and the copyright header

File: host/xdc/event_generator.go (hunk 2 above)
Comment (yux0) [reply host/xdc/event_generator.go]: Added more detail about the interfaces.

File: host/xdc/history_event_test.go (hunk 3 above)
Comment (yux0) [reply host/xdc/history_event_test.go]: Replace all the short names with readable names

**Pull request merged**